src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -184,6 +193,27 @@ public static bool IsDisposeImplementation([NotNullWhen(returnValue: true)] this
                 method.IsImplementationOfInterfaceMethod(null, iDisposable, "Dispose");
         }
 
+        /// <summary>
+        /// Checks if the given method implements "IAsyncDisposable.Dispose" or overrides an implementation of "IAsyncDisposable.Dispose".
+        /// </summary>
+        public static bool IsAsyncDisposeImplementation([NotNullWhen(returnValue: true)] this IMethodSymbol? method, [NotNullWhen(returnValue: true)] INamedTypeSymbol? iAsyncDisposable)
+        {
+            if (method == null)
+            {
+                return false;
+            }
+
+            if (method.IsOverride)
+            {
+                return method.OverriddenMethod.IsAsyncDisposeImplementation(iAsyncDisposable);
+            }
+
+            // Identify the implementor of IAsyncDisposable.Dispose in the given method's containing type and check
+            // if it is the given method.
+            return method.Parameters.Length == 0 &&
+                method.IsImplementationOfInterfaceMethod(null, iAsyncDisposable, "DisposeAsync");
```

Comment:
Is checking the method return type necessary here?
If so, please tell me how to write it.

Reply:
I would say that's probably better to check the return type. You will need to do the following `method.ReturnType.Equals(valueTaskType)` and to get `valueTaskType` you will need to either add a parameter of type `INamedTypeSymbol` to this method declaration to delegate getting the type to the caller or you could mimic what's done on `IsAsyncDisposeImplementation` (i.e. `INamedTypeSymbol? valueTaskType= compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemValueTask);`).